localstack/services/kms/provider.py

```diff
@@ -133,182 +159,336 @@ def __init__(self, message=None):
-
-        return result
+        key = KmsKey(create_key_request, context.account_id, context.region)
+        key_id = key.metadata.get("KeyId")
+        self._get_store(context).keys[key_id] = key
+        return CreateKeyResponse(KeyMetadata=key.metadata)
+
+    @handler("ScheduleKeyDeletion", expand=False)
+    def schedule_key_deletion(
+        self, context: RequestContext, schedule_key_deletion_request: ScheduleKeyDeletionRequest
+    ) -> ScheduleKeyDeletionResponse:
+        pending_window = int(schedule_key_deletion_request.get("PendingWindowInDays", 30))
+        if pending_window < 7 or pending_window > 30:
+            raise ValidationException(
+                f"PendingWindowInDays should be between 7 and 30, but it is {pending_window}"
+            )
+        key = _get_key(schedule_key_deletion_request.get("KeyId"), self._get_store(context))
+        key.schedule_key_deletion(pending_window)
+        attrs = ["DeletionDate", "KeyId", "KeyState"]
+        result = select_attributes(key.metadata, attrs)
+        result["PendingWindowInDays"] = pending_window
+        return ScheduleKeyDeletionResponse(**result)
+
+    @handler("CancelKeyDeletion", expand=False)
+    def cancel_key_deletion(
+        self, context: RequestContext, cancel_key_deletion_request: CancelKeyDeletionRequest
+    ) -> CancelKeyDeletionResponse:
+        key = _get_key(cancel_key_deletion_request.get("KeyId"), self._get_store(context))
+        key.metadata["KeyState"] = "Disabled"
+        key.metadata["DeletionDate"] = None
+        # https://docs.aws.amazon.com/kms/latest/APIReference/API_CancelKeyDeletion.html#API_CancelKeyDeletion_ResponseElements
+        # "The Amazon Resource Name (key ARN) of the KMS key whose deletion is canceled."
+        return CancelKeyDeletionResponse(KeyId=key.metadata.get("Arn"))
+
+    @handler("DisableKey", expand=False)
+    def disable_key(self, context: RequestContext, disable_key_request: DisableKeyRequest) -> None:
+        key = _get_key(disable_key_request.get("KeyId"), self._get_store(context))
+        key.metadata["KeyState"] = "Disabled"
+        key.metadata["Enabled"] = False
+
+    @handler("EnableKey", expand=False)
+    def enable_key(self, context: RequestContext, enable_key_request: EnableKeyRequest) -> None:
+        key = _get_key(enable_key_request.get("KeyId"), self._get_store(context))
+        key.metadata["KeyState"] = "Enabled"
+        key.metadata["Enabled"] = True
+
+    @handler("ListKeys", expand=False)
+    def list_keys(
+        self, context: RequestContext, list_keys_request: ListKeysRequest
+    ) -> ListKeysResponse:
+        # https://docs.aws.amazon.com/kms/latest/APIReference/API_ListKeys.html#API_ListKeys_ResponseSyntax
+        # Out of whole KeyMetadata only two fields are present in the response.
+        keys_list = PaginatedList(
+            [
+                {"KeyArn": key.metadata.get("KeyArn"), "KeyId": key.metadata.get("KeyId")}
```

Comment:
nit: we could also use `select_attributes(..)` here (as further above):
```suggestion
                select_attributes(key.metadata, ("KeyArn", "KeyId"))
```

Reply:
Done.

Reply:
Got back to my original approach, as what is KeyArn in the response is just Arn in the metadata, so can't just easily select it. And selecting only one field while adding the other outside of the select_attributes() doesn't make sense.